### PR TITLE
Validate Input Policy and show info on user code caches

### PR DESCRIPTION
**Pull request by shubham3121**
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

**Comment by shubham3121** [comment]
Caching Warning and propagating error message.

Screenshot from 2024-03-18 18-46-51

**Comment by shubham3121** [comment]
Propagating error when Input policy is not longer valid (e.g. running a function on a different dataset)

Screenshot from 2024-03-18 19-08-43

File: packages/syft/src/syft/service/action/action_service.py [hunk 1]
```diff
@@ -302,45 +302,34 @@ def _user_code_execute(
...
                 if not code_item.output_policy_approved:
                     return Err("Execution denied: Your code is waiting for approval")
-                return Err(f"No input poliicy defined for user code: {code_item.id}")
+                return Err(f"No input policy defined for user code: {code_item.id}")
+
+            # Filter input kwargs based on policy
             filtered_kwargs = input_policy.filter_kwargs(
                 kwargs=kwargs, context=context, code_item_id=code_item.id
             )
-            if isinstance(filtered_kwargs, SyftError) or filtered_kwargs.is_err():
```
Comment (teo-milea) [inline packages/syft/src/syft/service/action/action_service.py:314]: so glad to see we are slowly removing the SyftError type for the internal API!

File: packages/syft/src/syft/protocol/protocol_version.json
Comment (teo-milea) [inline packages/syft/src/syft/protocol/protocol_version.json:1]: Not sure how the protocol versioning works right now, but why do we have so many changes without changes to objects?

**Review by teo-milea — APPROVED** [review]
Great work!

**Pull request merged**